pkg/identity/cache/cache.go

```diff
@@ -184,6 +195,13 @@ func (w *identityWatcher) watch(events allocator.AllocatorEventRecvChan) {
 			}
 			// Issue collected updates
 			w.owner.UpdateIdentities(added, deleted) // disjoint sets
+
+			if synced && !closed {
+				w.logger.Info("All initial identities added to SelectorCache.")
+				close(w.watchDone)
+				closed = true
+			}
```

Comment:
I would suggest adding a comment that it is not necessary to wait on the channel returned by `UpdateIdentities`, because it is enough that the SelectorCache is up to date to unblock regeneration, as the policy maps would be updated at that point anyways (assuming I understood this logic correctly).